solidity/test/isms/OptimismISM.t.sol

```diff
@@ -0,0 +1,461 @@
+// SPDX-License-Identifier: MIT or Apache-2.0
+pragma solidity ^0.8.13;
+
+import {Test} from "forge-std/Test.sol";
+import "forge-std/console.sol";
+
+import {TypeCasts} from "../../contracts/libs/TypeCasts.sol";
+import {Mailbox} from "../../contracts/Mailbox.sol";
+import {Message} from "../../contracts/libs/Message.sol";
+import {TestMultisigIsm} from "../../contracts/test/TestMultisigIsm.sol";
+import {OptimismISM} from "../../contracts/isms/native/OptimismISM.sol";
+import {OptimismMessageHook} from "../../contracts/hooks/OptimismMessageHook.sol";
+import {TestRecipient} from "../../contracts/test/TestRecipient.sol";
+
+import {Lib_CrossDomainUtils} from "@eth-optimism/contracts/libraries/bridge/Lib_CrossDomainUtils.sol";
+import {AddressAliasHelper} from "@eth-optimism/contracts/standards/AddressAliasHelper.sol";
+import {ICrossDomainMessenger} from "@eth-optimism/contracts/libraries/bridge/ICrossDomainMessenger.sol";
+import {ICanonicalTransactionChain} from "@eth-optimism/contracts/L1/rollup/ICanonicalTransactionChain.sol";
+import {L2CrossDomainMessenger} from "@eth-optimism/contracts-bedrock/contracts/L2/L2CrossDomainMessenger.sol";
+import {Encoding} from "@eth-optimism/contracts-bedrock/contracts/libraries/Encoding.sol";
+import {Hashing} from "@eth-optimism/contracts-bedrock/contracts/libraries/Hashing.sol";
+
+contract OptimismISMTest is Test {
+    uint256 internal mainnetFork;
+    uint256 internal optimismFork;
+
+    address internal constant L1_MESSENGER_ADDRESS =
+        0x25ace71c97B33Cc4729CF772ae268934F7ab5fA1;
+    address internal constant L1_CANNONICAL_CHAIN =
+        0x5E4e65926BA27467555EB562121fac00D24E9dD2;
+    address internal constant L2_MESSENGER_ADDRESS =
+        0x4200000000000000000000000000000000000007;
+
+    uint8 internal constant VERSION = 0;
+    uint256 internal constant DEFAULT_GAS_LIMIT = 1_920_000;
+
+    address internal alice = address(0x1);
+
+    ICrossDomainMessenger internal l1Messenger;
+    L2CrossDomainMessenger l2Messenger;
+    OptimismISM internal opISM;
+    OptimismMessageHook internal opHook;
+
+    TestRecipient internal testRecipient;
+    bytes internal testMessage =
+        abi.encodePacked("Hello from the other chain!");
+
+    uint32 internal constant MAINNET_DOMAIN = 1;
+    uint32 internal constant OPTIMISM_DOMAIN = 10;
+
+    event SentMessage(
+        address indexed target,
+        address sender,
+        bytes message,
+        uint256 messageNonce,
+        uint256 gasLimit
+    );
+
+    event RelayedMessage(bytes32 indexed msgHash);
+
+    event FailedRelayedMessage(bytes32 indexed msgHash);
+
+    event ReceivedMessage(address indexed sender, bytes32 indexed messageId);
+
+    error NotCrossChainCall();
+
+    function setUp() public {
+        mainnetFork = vm.createFork(vm.rpcUrl("mainnet"));
+        optimismFork = vm.createFork(vm.rpcUrl("optimism"));
```

Comment:
in hindsight, unit tests without forking are probably preferred but no need to undo this work now